### ISSUE TITLE
get_var() returns NULL when empty string has been found

### ISSUE THREAD

**Issue by flymikeGit** [WordPress/Documentation-Issue-Tracker#1671]
## Issue Description

Description of Return values is "string|null Database query result (as string), or null on failure.". But, per https://core.trac.wordpress.org/ticket/30257, NULL will also be returned if the **_successful_** result is an empty string.
## URL of the Page with the Issue]
https://developer.wordpress.org/reference/classes/wpdb/get_var/

## Section of Page with the issue

Return values
## Why is this a problem?

When my code found a return value of NULL, it triggered my error processing logic - instead of returning a successful result of an empty string.
## Suggested Fix

Make clear that a NULL return **_may_** indicate an error, but could also be the successful result of an empty string.

**Comment by JakeQZ** [comment]
I don't think this is merely a documentation issue.  Surely if the value in the database is an empty string, then an empty string should be returned.  If it is `NULL` (for a nullable value) then `null` should be returned.  In case of an error, it would perhaps be best for `false` to be returned, as, somewhat fortunately, SQL does not support Boolean values.  Ideally upon error, an exception would be thrown, but I don't think that would fit in well with the current WordPress framework which doesn't generally throw or handle exceptions.

I came across this issue using the [Custom Product Tabs for WooCommerce](https://wordpress.org/plugins/yikes-inc-easy-custom-woocommerce-product-tabs/) which seems to have (probably erroneously) inserted a `postmeta` table row with empty strings (*not* `NULL` values) for `meta_key` and `meta_value`.  This leads to a depreaction notice with PHP 8.1 ("preg_replace(): Passing null to parameter #&#8203;3 ($subject) of type array|string is deprecated") in `is_protected_meta()` called from `meta_form()` because the `$keys` array obtained via `$wpdb->get_col()` contains `null` where it should have an empty string.

I think these issues will need fixing for PHP 9.0, when the deprecation notices will become fatal errors.

> ## Suggested Fix
> 
> Make clear that a NULL return **_may_** indicate an error, but could also be the successful result of an empty string.

## Suggested Fix

- Return an empty string when the value is an empty string;
- Return `null` when the value is `NULL`;
- Return `false` upon error (or throw an exception).
